dChatServer/ChatPacketHandler.cpp

```diff
@@ -337,121 +337,142 @@ void ChatPacketHandler::HandleRemoveFriend(Packet* packet) {
 
 void ChatPacketHandler::HandleChatMessage(Packet* packet) {
 	CINSTREAM_SKIP_HEADER;
-	LWOOBJID playerID = LWOOBJID_EMPTY;
+	LWOOBJID playerID;
 	inStream.Read(playerID);
 
 	const auto& sender = Game::playerContainer.GetPlayerData(playerID);
+	if (!sender || sender.GetIsMuted()) return;
 
-	if (!sender) return;
-
-	if (sender.GetIsMuted()) return;
-
-	inStream.SetReadOffset(0x14 * 8);
-
-	uint8_t channel = 0;
+	eChatChannel channel;
+	uint32_t size;
+	
+	inStream.IgnoreBytes(4);
 	inStream.Read(channel);
+	inStream.Read(size);
+	inStream.IgnoreBytes(77);
 
-	std::string message = PacketUtils::ReadString(0x66, packet, true, 512);
-
-	LOG("Got a message from (%s) [%d]: %s", sender.playerName.c_str(), channel, message.c_str());
+	LUWString message(size);
+	inStream.Read(message);
+	
+	LOG("Got a message from (%s) via [%s]: %s", sender.playerName.c_str(), StringifiedEnum::ToString(channel).data(), message.GetAsString().c_str());
 
-	if (channel != 8) return;
+	if (channel == eChatChannel::TEAM) {
```

Comment:
Why was this inverted?  Bouncer was good for early return before to reduce scope.  If its to allow for more channels, consider a switch case instead for future use intent

Reply:
yeah, there can be more channels supported, so a switch case would work well here